src/components/shared/chart/Canvas.js

```diff
@@ -368,10 +368,11 @@ export class ChartCanvas<Item> extends React.Component<
         // selected item can get out of sync. Invalidate it to make sure that
         // it's always fresh. This setState will cause a rerender, but we have
         // to do it to prevent any crashes or incorrect tooltip positions.
+        // This is okay to do it because the main `prevProps !== this.props`
+        // check above will return false and will not schedule additional drawing.
         this.setState({ selectedItem: null });
-      } else {
-        this._scheduleDraw();
       }
+      this._scheduleDraw();
```

Comment:
So previously we were only updating the canvas when the `previous props !== new props` AND `selectedItem state was either null or not the same` (to clear the selectedItem state). I initially thought that the `setState` above would create another re-render and draw eventually. But on the second render after the `setState`, it's `previous props !== new props` doesn't return true anymore and doesn't render. So my initial assumption of drawing on the second render didn't work. 

Now we're scheduling a draw whenever the previous props are not the same with the new props. This makes sure that we always have the consistent canvas. And on the re-render it never goes into this if branch, so it doesn't cost us a lot (only a single canvas element render, but that should be okay).